src/changelogs/fetch_changelogs.py

```diff
@@ -4,27 +4,55 @@
 import feedparser
 from bs4 import BeautifulSoup
 from urllib import request
+from utils import json_utils
 
 
 class ChangelogFetcher:
     def __init__(self):
-        self.changelogs_by_date = {}
+        self.changelogs_by_version = {}
+        # layer1: version # as assigned by developers
+        # layer2: content (reworked in another pr, not gonna bother writing temporary
+        # stuff here for layer2-4)
+
+        self.changelog_date_links = {}
+        # layer1: version # as assigned by developers
+        # layer2: date
 
     def get_rss(self, rss_url, update_existing=False):
         self.RSS_URL = rss_url
         self._fetch_forum_changelogs(update_existing)
-        return self.changelogs_by_date
+
+        return self.changelogs_by_version
 
     def get_txt(self, changelog_path):
         self._process_local_changelogs(changelog_path)
-        return self.changelogs_by_date
+        return self.changelogs_by_version
 
     def changelogs_to_file(self, output_dir):
-        for date, changelog in self.changelogs_by_date.items():
-            os.makedirs(output_dir, exist_ok=True)
-            with open(output_dir + f'/{date}.txt', 'w', encoding='utf8') as f_out:
+        for version, changelog in self.changelogs_by_version.items():
+            raw_output_dir = os.path.join(output_dir, 'raw')
+            os.makedirs(raw_output_dir, exist_ok=True)
+            with open(raw_output_dir + f'/{version}.txt', 'w', encoding='utf8') as f_out:
                 f_out.write(changelog)
 
+        # Read existing changelogs.json content,
+        # add any keys that are not yet present or have differing values,
+        # write back
+
+        # this file is not overwritten even when update_existing is True
+        # many entries were manually added due to only the first page on the site has rss feed
+        changelogs_path = output_dir + '/changelogs.json'
+        existing_changelogs = json_utils.read(changelogs_path)
+        existing_changelogs.update(self.changelog_date_links)
+
+        # Sort existing_changelogs by key but numerically
+        keys = list(existing_changelogs.keys())
+        # Sort the keys numerically, not lexiconically
```

Comment:
Lexiconically??

This comment can just go, `.sort()` is pretty self explanitory

Reply:
I think its worthwhile, though I meant lexicographically. 

Lexicographical sort:
1, 10, 100, 11, 12

Numerical sort:
1, 2, 3, ... 10, 11

Reply:
damn that's a new word to add to my.... lexicon